spirv_msl.cpp

```diff
@@ -5485,14 +5864,30 @@ void CompilerMSL::emit_instruction(const Instruction &instruction)
 		uint32_t a = ops[2];
 		uint32_t b = ops[3];
 
+		auto *type_a = maybe_get<SPIRType>(a);
+		auto *type_b = maybe_get<SPIRType>(b);
+
 		auto &type = get<SPIRType>(result_type);
 		string expr = type_to_glsl_constructor(type);
 		expr += "(";
 		for (uint32_t col = 0; col < type.columns; col++)
 		{
-			expr += to_enclosed_expression(a);
-			expr += " * ";
-			expr += to_extract_component_expression(b, col);
+			if (msl_options.invariant_float_math && type_a && type_b &&
+			    ((is_matrix(*type_a) && is_matrix(*type_b)) || (is_matrix(*type_a) && is_vector(*type_b)) ||
+			     (is_vector(*type_a) && is_matrix(*type_b))))
+			{
+				expr += "spvFMulMatrixMatrix(";
```

Comment:
Is this correct? MatrixMatrix implies 4x4 matrix multiply, this is a vector times scalar thing.

Reply:
This was either a sanity check or simply a leftover before `OpVectorTimesMatrix` differentiation. I removed that now and also added a new test case as well as `--msl-invariant-float-math` compiler option.